framework/src/service/ServiceReferenceBasePrivate.cpp

```diff
@@ -82,14 +84,14 @@ namespace cppmicroservices
                 }
             }
             {
-                auto l = registration->properties.Lock();
+                auto l = coreInfo->properties.Lock();
                 US_UNUSED(l);
                 for (auto const& clazz : ref_any_cast<std::vector<std::string>>(
-                         registration->properties.ValueByRef_unlocked(Constants::OBJECTCLASS)))
+                         coreInfo->properties.ValueByRef_unlocked(Constants::OBJECTCLASS)))
                 {
                     if (smap->find(clazz) == smap->end() && clazz != "org.cppmicroservices.factory")
                     {
-                        if (auto bundle_ = registration->bundle.lock())
+                        if (auto bundle_ = coreInfo->bundle_.lock())
```

Comment:
the initialization statements are still in. is this intended?

Reply:
This actually is a bool-convertible condition expression, not the C++17-only `if (init-statement ; condition)`. I also was not consciously aware that this is legal C++14, but for these PRs I only fix things the compilers complains about in C++14-mode.

TLDR; all is fine, this is legal C++14